src/main.rs

```diff
@@ -33,38 +33,55 @@ pub fn main() -> Result<()> {
 
     let (args, state_file) = cli::init_cli();
 
-    // setup platform (in our case, PC)
-    let trussed_platform = platform::init_platform(state_file);
-
-    // setup Trussed
-    let mut trussed_service = trussed::service::Service::new(trussed_platform);
-    let client_id = "totp";
-    // In real life, `trussed_service.try_new_client` has an additional parameter that is a `Syscall`
-    // implementation; giving the client a way to signal the ambient runtime to call the service.
-    // Here, we use the service's implementation of `Syscall`, where it simply calls itself :)
-    let trussed_client = trussed_service.try_as_new_client(client_id).unwrap();
-
-    // setup authenticator
-    let mut authenticator = authenticator::Authenticator::new(trussed_client);
-
+    //setup wireguard
+    let trussed_platform_wg = platform::init_platform(state_file.clone());
+    let mut trussed_service_wg = trussed::service::Service::new(trussed_platform_wg);
+    let client_id_wg = "wireguard";
+    let trussed_client_wg = trussed_service_wg.try_as_new_client(client_id_wg).unwrap();
+    let mut wireguard = wireguard::Wireguard::new(trussed_client_wg);
 
+    
     // The "runner"'s actual "scheduling" part starts here
     info!("Let's go!");
 
+
+
     // the "args" come in over the CLI "interface", and are "deserialized" for processing
     // using `Command`'s implementation of `TryFrom`, the standard Trait for fallible type conversion
-    let command = authenticator::Command::try_from(&args)?;
+    let wg_command = wireguard::WgCommand::try_from(&args)?;
+
 
     // the command is "dispatched" into the application
-    match command {
-        authenticator::Command::Register(register) => {
-            authenticator.register(&register)?;
+    match wg_command {
+       wireguard::WgCommand::Unlock(unlock) => 
+        {
```

Comment:
A pattern I use here is bringing all the enum variants in scope:
```
use wireguard::WgCommand::*;
match command {
  Unlock(unlock) => (...)
```

Also, generally it's a C-ism to use prefixes like `wg_` or `Wg`. When the `WgCommand` is already in the `wireguard` module, it's clear that `wireguard::Command` refers to a WireGuard command. If you want to import without the module path prefix, you can do that at call site: `use wireguard::Command as WgCommand`.